Source/Core/CameraLimiter.js

```diff
@@ -0,0 +1,148 @@
+define([
+    '../Core/AxisAlignedBoundingBox',
+    '../Core/BoundingRectangle',
+    '../Core/BoundingSphere',
+    '../Core/Cartesian2',
+    '../Core/Cartesian3',
+    '../Core/Cartesian4',
+    '../Core/Cartographic',
+    '../Core/Check',
+    '../Core/defaultValue',
+    '../Core/defined',
+    '../Core/defineProperties',
+    '../Core/DeveloperError',
+    '../Core/EasingFunction',
+    '../Core/Ellipsoid',
+    '../Core/EllipsoidGeodesic',
+    '../Core/Event',
+    '../Core/HeadingPitchRange',
+    '../Core/HeadingPitchRoll',
+    '../Core/Intersect',
+    '../Core/IntersectionTests',
+    '../Core/Math',
+    '../Core/Matrix3',
+    '../Core/Matrix4',
+    '../Core/OrientedBoundingBox'
+], function(
+    AxisAlignedBoundingBox,
+    BoundingRectangle,
+    BoundingSphere,
+    Cartesian2,
+    Cartesian3,
+    Cartesian4,
+    Cartographic,
+    Check,
+    defaultValue,
+    defined,
+    defineProperties,
+    DeveloperError,
+    EasingFunction,
+    Ellipsoid,
+    EllipsoidGeodesic,
+    Event,
+    HeadingPitchRange,
+    HeadingPitchRoll,
+    Intersect,
+    IntersectionTests,
+    CesiumMath,
+    Matrix3,
+    Matrix4,
+    OrientedBoundingBox) {
+    'use strict';
+
+    /**
+     * The CameraLimiter is used to limit a {@Camera}'s position (using a bounding object) and orientation (defined by {@HeadingPitchRange}.
+     *
+     * @alias CameraLimiter
+     * @constructor
+     *
+     * @param {AxisAlignedBoundingBox|BoundingRectangle|BoundingSphere|OrientedBoundingBox} [options.boundingObject] Assigned to the boundingObject attribute which can be used to specify limits on locations.
+     * @param {HeadingPitchRoll} [options.minimumHeadingPitchRoll] Assigned to the minHeadingPitchRoll attribute which can be used to limit the minimum orientation.
+     * @param {HeadingPitchRoll} [options.maximumHeadingPitchRoll] Assigned to the maxHeadingPitchRoll attribute which can be used to limit the maximum orientation.
+     */
+    function CameraLimiter(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * The BoundingObject.
+         * @type {AxisAlignedBoundingBox|BoundingRectangle|BoundingSphere|OrientedBoundingBox}
+         * @default undefined
+         */
+        this.boundingObject = options.boundingObject;
+
+        /**
+         * The minimum orientation values.
+         * @type {HeadingPitchRoll}
+         * @default undefined
+         */
+        this.minHeadingPitchRoll = HeadingPitchRoll.clone(options.minimumHeadingPitchRoll);
```

Comment:
What if I wanted to limit the `heading`, but not the `pitch` and `roll`?  Is there a way to do that?

Reply:
@hpinkos To do this a user would just set whichever minheading/pitch/roll maxheading/pitch/roll quantity they wished to be restricted and allow the rest to be Infinity/negInfinity